File: server/src/models/Restaurant.js
const mongoose = require('mongoose');

const restaurantSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  nameAr: {
    type: String,
    required: true,
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  descriptionAr: {
    type: String,
    trim: true
  },
  image: {
    type: String,
    required: true
  },
  coverImage: {
    type: String
  },
  rating: {
    type: Number,
    default: 4.0,
    min: 0,
    max: 5
  },
  reviewCount: {
    type: Number,
    default: 0
  },
  deliveryTime: {
    type: String,
    required: true,
    default: '30-40 دقيقة'
  },
  deliveryFee: {
    type: Number,
    default: 0
  },
  minimumOrder: {
    type: Number,
    default: 50
  },
  isOpen: {
    type: Boolean,
    default: true
  },
  isActive: {
    type: Boolean,
    default: true
  },
  isFavorite: {
    type: Boolean,
    default: false
  },
  isTopRated: {
    type: Boolean,
    default: false
  },
  categories: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Category'
  }],
  location: {
    type: {
      type: String,
      enum: ['Point'],
      default: 'Point'
    },
    coordinates: {
      type: [Number],
      default: [0, 0]
    }
  },
  address: {
    type: String,
    required: true
  },
  phone: {
    type: String,
    required: true
  },
  owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  }
}, {
  timestamps: true
});

restaurantSchema.index({ location: '2dsphere' });
restaurantSchema.index({ isActive: 1, isOpen: 1 });
restaurantSchema.index({ isFavorite: 1 });
restaurantSchema.index({ isTopRated: 1 });
restaurantSchema.index({ rating: -1 });

module.exports = mongoose.model('Restaurant', restaurantSchema);

File: lib/features/category/data/model/restaurant.dart
class Restaurant {
  final String id;
  final String name;
  final String image;
  final double rating;
  final int reviewsCount;
  final String deliveryTime;
  final int deliveryFee;
  final int? originalDeliveryFee;
  final String cuisine;
  final String? discount;
  final bool isFavorite;
  final bool isPromoted;
  final List<String> tags;
  final int minimumOrder;

  Restaurant({
    required this.id,
    required this.name,
    required this.image,
    required this.rating,
    required this.reviewsCount,
    required this.deliveryTime,
    required this.deliveryFee,
    this.originalDeliveryFee,
    required this.cuisine,
    this.discount,
    required this.isFavorite,
    this.isPromoted = false,
    required this.tags,
    required this.minimumOrder,
  });
}
